File: src/main/java/com/dolts/hotelaria/models/HotelModel.java
package com.dolts.hotelaria.models;

import jakarta.persistence.*;
import lombok.*;

import java.io.Serializable;
import java.util.List;

@Entity
@Table(name = "hotel")
@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
@EqualsAndHashCode
public class HotelModel implements Serializable {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    private String nome;
    @Column(nullable = false, unique = true, length = 14)
    private String cnpj;
    private String responsavel;
    private String cpfResponsavel;
    @OneToMany(cascade = CascadeType.ALL, fetch = FetchType.LAZY, orphanRemoval = true)
    private List<EnderecoModel> endereco;
}

File: hotelaria-frontend/src/app/shared/models/hotel.model.ts
import {EnderecoModel} from "./endereco.model";

export class HotelModel {
    id: number;
    nome: string;
    cnpj: string;
    responsavel: string;
    cpfResponsavel: string;
    endereco: EnderecoModel[];
    logoMarcaId: number;
    logoAsDataSource: string | null | undefined;
    versao: number;
}
